Source/Core/VideoCommon/LightingShaderGen.h

```diff
@@ -62,7 +62,8 @@ static void GenerateLightShader(T& object, LightingUidData& uid_data, int index,
 		case LIGHTATTN_NONE:
 		case LIGHTATTN_DIR:
 			object.Write("ldir = normalize(" LIGHT_POS".xyz - pos.xyz);\n", LIGHT_POS_PARAMS(index));
-			object.Write("attn = 1.0f;\n");
+			object.Write("attn = 1.0;\n");
+			object.Write("if (length(ldir) == 0.0)\n\t ldir = float3(1.0, 1.0, 1.0);\n");
```

Comment:
Oh, so we're not dividing by zero, we're dividing by infinity?  That's not exactly comforting, but I guess it's a bit more predictable.  A comment describing that would be nice.

My instinct would probably be to conditionally scale the vector to avoid the overflow (`if (vec.x > 0x1p60 || vec.y > 0x1p60 || vec.z > 0x1p60) vec /= 0x1p80`); that way, you actually get the correct normalized vector.  Your choice, though.